Practical7/src/store/index.js

```diff
@@ -0,0 +1,173 @@
+import { createStore } from "vuex";
+import axios from "axios";
+import router from "../router";
+
+const store = createStore({
+  state: {
+    allCars: [],
+    loading: null,
+    apiError: null,
+
+    userAuth: JSON.parse(localStorage.getItem("userAuth")),
+
+    formDetails: {
+      showForm: false,
+      formHeading: "",
+      formAction: "",
+      formInitialValue: {
+        id: 0,
+        name: "",
+        details: "",
+        image: "",
+        price: null,
+      },
+    },
+  },
+  mutations: {
+    //This is used for Synchronous
+    setCars(state, payload) {
+      state.allCars = payload;
+    },
+    setUserAuth(state, payload) {
+      state.userAuth = payload;
+    },
+    setForm(state, payload) {
+      state.formDetails = payload;
+    },
+    setLoading(state, payload) {
+      state.loading = payload;
+    },
+    setApiError(state, payload) {
+      state.apiError = payload;
+    },
+  },
+  getters: {
+    getUserAuth: (state) => state.userAuth,
+    getForm: (state) => state.formDetails,
+    getCars: (state) => state.allCars,
+    getLoading: (state) => state.loading,
+    getApiError: (state) => state.apiError,
+  },
+  actions: {
+    // This is used for Asunchronous
+    setLoading: ({ commit }, state, payload) =>
+      commit("setLoading", state, payload),
+    setApiError: ({ commit }, state, payload) =>
+      commit("setApiError", state, payload),
+    setUserAuth: ({ commit }, state, payload) =>
+      commit("setUserAuth", state, payload),
+    setForm: ({ commit }, state, payload) => commit("setForm", state, payload),
+
+    async fetchCarDetails({ commit }) {
+      commit("setLoading", true);
+      await axios
+        .get("https://testapi.io/api/dartya/resource/cardata")
+        .then((res) => {
+          commit("setCars", res.data.data);
+          commit("setLoading", false);
+        })
+        .catch((err) => {
+          commit("setApiError", err);
+          commit("setLoading", false);
+        });
+    },
+
+    async postCarDetails({ commit, dispatch }, payload) {
+      commit("setLoading", true);
+      await axios
+        .post("https://testapi.io/api/dartya/resource/cardata", payload)
+        .then((res) => {
+          if (res.status === 201) {
+            dispatch("fetchCarDetails");
+          }
+          commit("setLoading", false);
+          commit("setForm", (this.state.formDetails.showForm = false));
+        })
+        .catch((err) => {
+          commit("setLoading", false);
+          commit("setForm", (this.state.formDetails.showForm = false));
+          alert("can't do the operation at this moment" + err);
```

Comment:
use toast message instead of alert

Reply:
Implemented on next vuejs practical 8 as we discussed.